### ISSUE TITLE
nameservers resolved in wrong order

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2290]
|<img alt="ilia's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| ilia reported|
|-|-|
|Trac ID|trac#2290|
|Type|defect|
|Created|2006-12-11 02:36:36Z|

If there are multiple addresses in resolv.conf, they are queried bottom-up, last server is queried first.

Although, the order should be top-down, by "man resolv.conf", and other programs also use top-down order: 
first server is queried first.

Probably, that can be achieved by putting pickServer into queryUDP, but the single-line attached patch uses simpler approach.

Attachments:

* [resolv.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/719/71942625febf9aa13b56f19e0bc5c34c4039f1a7/ec10ebc861b5c815af159449365d6b6267c34df3.patch) (748 bytes) - added by ilia on 2006-12-11 02:36:54Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__2290 2290
type__defect defect
reporter__ilia ilia
priority__highest highest
milestone__Names_0_4 Names-0.4
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__names names
keywords__ 
time__1165804596000000 1165804596000000
changetime__1186492917000000 1186492917000000
version__None None
owner__ 
cc__therve
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="ilia's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_1"></a>|ilia commented|
|-|-|

Wops, the patch seem to reverse whole address, but only first part should be reversed before adding dynamic servers to list.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun removed owner|
|-|-|

Done in resolver-query-order-2290.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve set owner to @exarkun|
|-|-|

 * update copyright of both files
 * whitespace cleanups

Then please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [20977]) Merge resolver-query-order-2290

Author: exarkun, ilia
Reviewer: therve
Fixes [#2290](https://github.com/twisted/twisted/issues/2290)

Change twisted.names.client.Resolver to query specified name servers, both those
in the list given to its initialize and those from any supplied resolv.conf
filename, in the order they are specified, rather than in reverse order.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
